src/rgw/rgw_rest_pubsub.cc

```diff
@@ -88,6 +88,9 @@ class RGWPSCreateTopicOp : public RGWOp {
 
     dest.push_endpoint = s->info.args.get("push-endpoint");
     s->info.args.get_bool("persistent", &dest.persistent, false);
+    s->info.args.get_int("persistency_time_to_live", reinterpret_cast<int *>(&dest.persistency_time_to_live), 0);
```

Comment:
why the reinterpret cast?
don't  we have a `get_unsigned()` function?

Reply:
no we actually don't have one

Reply:
ok. we can probabloy add that, but if we allow -1 we can probably just change to int

Reply:
I think we can keep it this way, anyway signed or unsigned denotes the interpretation of the number